bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,778 @@
+    private List<FerroampChannelConfiguration> channelConfigSsoS1 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS2 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS3 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    int ssoIdSet = 0;
+    boolean establishedConnection = false;
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonDisCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            sendMQTT(requestCmdJsonAuto);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS0 = FerroampChannelConfiguration.getChannelConfigurationSsoS0();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        ferroampConfig = getConfigAs(FerroampConfiguration.class);
+        @SuppressWarnings("null")
+        final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            boolean thingReachable = true;
+
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+                refresh();
```

Comment:
This would possibly flip the Thing from unknown to online to offline.
I recommend to extract the `get/sendMQTT` method to a seperate class and throw custom exceptions if something fails.
This would make that seperate class responsible for communication with the remote mqtt service and would make this ThingHandler responsible to update the channels / thing status.

Reply:
Ongoing 20240607